rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1341,70 +1220,109 @@ public BooleanWrapper createNodeSource(String nodeSourceName, String infrastruct
         }
     }
 
-    protected BooleanWrapper createNodeSource(NodeSourceData nodeSourceData, boolean nodesRecoverable) {
-        String nodeSourceName = nodeSourceData.getName();
+    public void createNotActiveNodeSource(NodeSourceData persistedNodeSource) {
+        InfrastructureManager infrastructureManager = InfrastructureManagerFactory.create(persistedNodeSource);
+
+        NodeSourcePolicy notActivePolicy = NodeSourcePolicyFactory.createNotActive(persistedNodeSource.getPolicyType());
+
+        NodeSource nodeSource = new NodeSource(this.getUrl(),
+                                               persistedNodeSource.getName(),
+                                               infrastructureManager,
+                                               notActivePolicy,
+                                               (RMCore) PAActiveObject.getStubOnThis(),
+                                               this.monitoring,
+                                               persistedNodeSource.toNodeSourceDescriptor());
+
+        String nodeSourceName = nodeSource.getName();
+
+        undeployedNodeSources.put(nodeSourceName, nodeSource);
+
+        this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_DEFINED,
+                                                              persistedNodeSource.getProvider().getName(),
+                                                              nodeSourceName,
+                                                              nodeSource.getDescription(),
+                                                              persistedNodeSource.getProvider().getName(),
+                                                              nodeSource.getDescriptor().getStatus().toString()));
+
+        logger.info("Node source " + nodeSourceName + " has been successfully defined by " +
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Node source " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)